src/Aspire.Hosting.Azure.ContainerRegistry/ContainerRegistryReferenceAnnotation.cs

```diff
@@ -0,0 +1,23 @@
+#pragma warning disable ASPIRECOMPUTE001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
+
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+
+namespace Aspire.Hosting.Azure;
+
+/// <summary>
+/// Annotation that indicates a resource is using a specific container registry.
+/// </summary>
+/// <remarks>
+/// Initializes a new instance of the <see cref="ContainerRegistryReferenceAnnotation"/> class.
+/// </remarks>
+/// <param name="registry">The container registry resource.</param>
+public class ContainerRegistryReferenceAnnotation(IContainerRegistry registry) : IResourceAnnotation
```

Comment:
Should this be in the same place `IContainerRegistry` is defined? (i.e. Aspire.Hosting)